contracts/router/quoter/SwapQuoterV2.sol

```diff
@@ -183,6 +191,16 @@ contract SwapQuoterV2 is PoolQuoterV1, Ownable {
         }
     }
 
+    // ═════════════════════════════════════════════ GENERAL QUOTES V2 ═════════════════════════════════════════════════
+
+    /// @inheritdoc ISwapQuoterV2
+    function areConnectedTokens(LimitedToken memory tokenIn, address tokenOut) external view returns (bool) {
+        // Check if this is a request for an origin swap.
+        // These are given with the tokenIn.actionMask set to the full set of actions.
+        bool isOriginSwap = tokenIn.actionMask == ActionLib.allActions();
```

Comment:
curious, why is this the case?

Reply:
It's a convention currently used for calling `quoter.getAmountOut()`:

https://github.com/synapsecns/synapse-contracts/blob/2d2a25d10f3f03ffafb347f8d699887c4334ce9f/contracts/bridge/router/SynapseRouter.sol#L272-L274

https://github.com/synapsecns/synapse-contracts/blob/2d2a25d10f3f03ffafb347f8d699887c4334ce9f/contracts/bridge/router/SynapseRouter.sol#L307-L309

The reason is that for destination swaps we are constrained by what the bridge is able to do with the token. E.g. 
https://github.com/synapsecns/synapse-contracts/blob/2d2a25d10f3f03ffafb347f8d699887c4334ce9f/contracts/bridge/SynapseBridge.sol#L339
vs
https://github.com/synapsecns/synapse-contracts/blob/2d2a25d10f3f03ffafb347f8d699887c4334ce9f/contracts/bridge/SynapseBridge.sol#L448

But for origin chain, the swap happens before bridging of the token is initiated, therefore anything that Router Adapter is able to do with the token (e.g. all Actions) is possible.